src/fides/api/schemas/tcf.py

```diff
@@ -57,47 +67,71 @@ class EmbeddedLineItem(FidesSchema):
     name: str
 
 
-class EmbeddedPurpose(EmbeddedLineItem):
-    """Sparse details for an embedded purpose beneath a system or vendor section.  Read-only."""
+class CommonVendorFields(FidesSchema):
+    """Fields shared between the three vendor sections of the TCF Experience"""
 
-    legal_bases: List[str] = []
+    id: str
+    has_vendor_id: Optional[bool]
+    name: Optional[str]
+    description: Optional[str]
 
 
-class TCFDataCategoryRecord(FidesSchema):
-    """Details for data categories on systems: Read-only"""
+class TCFConsentVendorRecord(UserSpecificConsentDetails, CommonVendorFields):
```

Comment:
okay this is a nit, but we have `TCFPurposeConsentRecord` above, and `TCFConsentVendorRecord` below—this tripped me up a few times while integrating because it switches between `[tcf_entity][legal_basis]` vs `[legal_basis][tcf_entity]`. I think most of these follow `[legal_basis][tcf_entity]`, so maybe we can switch `TCFPurposeConsentRecord` --> `TCFConsentPurposeRecord`? (and then I'd have to update FE stuff too I think)

Reply:
so [tcf_entity][legal_basis], leave TCFPurposeConsentRecord as is, but switch TCFConsentVendorRecord -

Reply:
OK I've added this commit, let me know if this is too disruptive - https://github.com/ethyca/fides/pull/4201/commits/8933d7d22f95f7007db9b54a1c4657f5a0167f53